sktime/forecasting/naive.py

```diff
@@ -139,5 +146,35 @@ def _predict_last_window(self, fh, X=None, return_pred_int=False,
             fh_idx = fh.index_like(self.cutoff)
             return last_window[fh_idx]
 
-        elif self.strategy == "mean":
+        elif self.strategy == "mean" and self.sp is None:
             return np.repeat(np.nanmean(last_window), len(fh))
+
+        elif self.strategy == "mean" and self.sp is not None:
+            last_window = pd.DataFrame(data=last_window,
+                                       columns=['data'])
+            # computing last season's mean and imputing it into last season
+            for i in range(self.sp_):
+                if any(last_window.index % self.sp_ == i) is True:
+                    last_window.\
+                        at[last_window[last_window.index % self.sp_ == i].
+                            index[-1], 'data'] =\
+                        last_window[last_window.index % self.sp_ == i].mean()
+
+            # we need to replicate the last window if max(fh) is
+            # larger than sp,
+            # so that we still make forecasts by repeating the
+            # last value for that season,
+            # assume fh is sorted, i.e. max(fh) == fh[-1]
+            # only slicing all the last seasons into last_window
+            if fh[-1] > self.sp_:
+                reps = np.int(np.ceil(fh[-1] / self.sp_))
+                last_window =\
+                    np.tile(last_window['data'].tail(self.sp_).to_numpy(),
+                            reps=reps)
+            else:
+                last_window =\
+                    last_window['data'].tail(self.sp_).to_numpy()
+
+            # get zero-based index by subtracting the minimum
+            fh_idx = fh.index_like(self.cutoff)
+            return last_window[fh_idx]
```

Comment:
I don't think we have to convert it into a pd.DataFrame, using the `last_window` in the numpy format should work even better. Why not do something along these lines? 

```python
seasonal_mean = last_window.reshape(window_length // sp, sp).mean(axis=0)
reps = np.int(np.ceil(fh[-1] / sp))
fh_idx = fh.index_like(self.cutoff)
y_pred = np.tile(seasonal_mean, reps=reps)[fh_idx]
```

Reply:
@mloning Nice idea, but when window_length is 5 and sp is 3 reshape won't work for that

Reply:
To implement this window_length must be a multiple of sp

Reply:
np.pad works fine

Reply:
But don't we check for that anyway in advance? Yes, so either we ignore (truncate) the remainder if window length is not a multiple or we pad the series, ignoring it may be easier? But if you prefer padding we can also go for that, but I think it's somewhat unintuitive to use more data points for some seasons than for others, perhaps we should simply raise an error if window length is not a multiple of sp?

Reply:
But here we are padding with -1 as marker and replacing them such that the final mean wont be affected by the padded extra elements i.e we are adding extra points such that the mean remains the same as it should be even after reshaping, this is not sluggish as we are using numpy and it adds the flexibility that window length not being multiple of sp